### PR TITLE
Formatting: Avoid formatting multi-line string literals since it may change behavior. rdar://32135036

### PR DESCRIPTION
**Pull request by nkcsgexi**
Before we figure out how to indent them consistently without altering behavior, we should leave them alone.

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

File: lib/IDE/Formatting.cpp [hunk 1]
```diff
@@ -760,6 +769,12 @@ class FormatWalker : public SourceEntityWalker {
   }
 
   bool walkToExprPre(Expr *E) override {
+    if (E->getKind() == ExprKind::StringLiteral &&
+        SM.isBeforeInBuffer(E->getStartLoc(), TargetLocation) &&
+        SM.isBeforeInBuffer(TargetLocation,
+                            Lexer::getLocForEndOfToken(SM, E->getEndLoc()))) {
+      InStringLiteral = true;
+    }
```
Comment (akyrtzi) [inline lib/IDE/Formatting.cpp]: What about if the line contains a plain string literal say, the line is `"blah"`, will this disable indentation for it ?

Comment (nkcsgexi) [inline lib/IDE/Formatting.cpp]: It won't touch that case because `"` is not after the start loc of the string literal.

**Review by akyrtzi — APPROVED** [review]

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Pull request merged**